Remotion/Data/DomainObjects.UnitTests/Factories/BaseConfiguration.cs

```diff
@@ -44,17 +44,27 @@ public static ITypeDiscoveryService GetTypeDiscoveryService (params Assembly[] r
       var assemblyFinder = new CachingAssemblyFinderDecorator(new AssemblyFinder(rootAssemblyFinder, assemblyLoader));
       ITypeDiscoveryService typeDiscoveryService = new AssemblyFinderTypeDiscoveryService(assemblyFinder);
 
-      return FilteringTypeDiscoveryService.CreateFromNamespaceBlacklist(
+      var whitelistedNamespaces = new[]
+                                  {
+                                      "Remotion.Data.DomainObjects.UnitTests.Mapping.TestDomain.ReflectionBasedPropertyResolver",
+                                      "Remotion.Data.DomainObjects.UnitTests.TestDomain",
+                                      "Remotion.Data.DomainObjects.UnitTests.DataManagement.TestDomain",
+                                      "Remotion.Data.DomainObjects.UnitTests.MixedDomains.TestDomain",
+                                      "Remotion.Data.DomainObjects.UnitTests.Linq.TestDomain"
+                                  };
+      return new FilteringTypeDiscoveryService(
           typeDiscoveryService,
-          "Remotion.Data.DomainObjects.UnitTests.Mapping.TestDomain.Errors",
-          "Remotion.Data.DomainObjects.UnitTests.Mapping.TestDomain.Integration",
-          "Remotion.Data.DomainObjects.UnitTests.Mapping.TestDomain.RelationReflector",
-          "Remotion.Data.DomainObjects.UnitTests.Mapping.TestDomain.Validation",
-          "Remotion.Data.DomainObjects.UnitTests.Mapping.MappingReflectionIntegrationTests",
-          "Remotion.Data.DomainObjects.UnitTests.Security.TestDomain",
-          "Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SchemaGenerationTestDomain",
-          "Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SqlServer.IntegrationTests"
-          );
+          type =>
+          {
+            var @namespace = type.Namespace ?? string.Empty;
+            for (var i = 0; i < whitelistedNamespaces.Length; i++)
+            {
+              if (@namespace.StartsWith(whitelistedNamespaces[i]))
+                return true;
+            }
```

Comment:
```suggestion
            if (whitelistedNamespaces.Any(@namespace.StartsWith))
              return true;
```
Whould this be more natural to read?

Reply:
Yes, but I wanted to keep the `for` for performance considerations as it prevents the allocation of the `IEnumerator<T>`. This might be optimized away by the JIT but in this case I would take the readability hit.

Reply:
Yeah, but given this is just a onetime hit during the unit tests, I would suggest we go with readability over GC.

In general: all the one-time stuff in the Framework always goes with readability over GC impact (unless something is really problematic) and only calls that happens many times during a request (e.g. for each property access), need to be GC optimized. And in between, it depends.